Mathlib/RingTheory/DedekindDomain/FiniteAdeleRing.lean

```diff
@@ -361,6 +382,18 @@ instance : CoeFun (FiniteAdeleRing R K)
     (fun _ ↦ ∀ (v : HeightOneSpectrum R), adicCompletion K v) where
   coe a v := a.1 v
 
+section Algebra
+
+variable (R' L : Type*) [CommRing R'] [Field L] [Algebra R L] [IsDedekindDomain R']
+  [Algebra R' L] [IsFractionRing R' L] [Algebra K (FiniteAdeleRing R' L)] [IsFractionRing R L]
+  [Algebra K L]  [IsScalarTower K L (FiniteAdeleRing R' L)]
```

Comment:
Do you need all of these? cant you reuse some of the ones from above?

Reply:
You're right I am able to shorten this quite a bit.